test/array.cpp

```diff
@@ -159,6 +159,7 @@ TEST(array_fill_assign) {
 
   array<int, shape<dim<>, dim<>>> sparse;
   auto sparse_shape = make_shape(dim<>(-2, 5, 2), dim<>(4, 10));
+  sparse_shape.resolve();
```

Comment:
This was a bug fix, the old check just happened to work because -9 satisfied the check

Reply:
Could we now add some extra runtime assertions around the code to check for using unresolved strides in functions like flat_min, etc.?

I have a feeling we'll catch a few bugs over here...

(Can be a different PR)

Reply:
One advantage of the new approach is that any use of the unresolved strides in arithmetic will probably be caught by ubsan or the like. Adding asserts is not trivial to do, it can’t be done in just a few places unfortunately (I tried).